include/boost/geometry/strategies/geographic/buffer_join_round.hpp

```diff
@@ -71,52 +59,34 @@ public :
                 CalculationType
             >::type;
 
+        using helper = geographic_buffer_helper<FormulaPolicy, calc_t>;
+
         calc_t const lon_rad = get_as_radian<0>(vertex);
         calc_t const lat_rad = get_as_radian<1>(vertex);
-        calc_t const lon1_rad = get_as_radian<0>(perp1);
-        calc_t const lat1_rad = get_as_radian<1>(perp1);
-        calc_t const lon2_rad = get_as_radian<0>(perp2);
-        calc_t const lat2_rad = get_as_radian<1>(perp2);
-
-        // Calculate angles from vertex to perp1/perp2
-        auto const inv1 = inverse<calc_t>::apply(lon_rad, lat_rad, lon1_rad, lat1_rad, m_spheroid);
-        auto const inv2 = inverse<calc_t>::apply(lon_rad, lat_rad, lon2_rad, lat2_rad, m_spheroid);
-
-        // For a sharp corner, perpendicular points are nearly opposite and the
-        // angle between the two azimuths can be nearly 180, but not more.
-        calc_t const two_pi = geometry::math::two_pi<calc_t>();
-        bool const wrapped = inv2.azimuth < inv1.azimuth;
-        calc_t const angle_diff = wrapped
-            ? ((two_pi + inv2.azimuth) - inv1.azimuth)
-            : inv2.azimuth - inv1.azimuth;
-
-        if (angle_diff < 0 || angle_diff > geometry::math::pi<calc_t>())
+
+        calc_t first_azimuth;
+        calc_t angle_diff;
+        if (! helper::calculate_angles(lon_rad, lat_rad, perp1, perp2, m_spheroid,
+                                       angle_diff, first_azimuth))
         {
-            // Defensive check with asserts
-            BOOST_GEOMETRY_ASSERT(angle_diff >= 0);
-            BOOST_GEOMETRY_ASSERT(angle_diff <= geometry::math::pi<calc_t>());
             return false;
         }
 
+        static calc_t const two_pi = geometry::math::two_pi<calc_t>();
```

Comment:
This is probably out of scope of this PR but we could replace `static` variables with `constexpr` variables. For this to work math constants would have to be `constexpr` functions.

Reply:
Indeed, I'll do that later in a separate PR